### PR TITLE
Fix For Unnatural Hips Behaviour When Sitting In HMD

### PR DESCRIPTION
**Pull request by amantley**
This PR addresses the bug addresses the bug [16982](https://highfidelity.manuscript.com/f/cases/16982/Sitting-in-HMD-Results-in-Impossible-Pose) in FogBugz.

A reset was added when the user's hips go below the default height of the hips by 1/3 of the default length of the leg.   

This means that when the user is sitting in real life their avatar will be standing.  

Also included are the hotfix changes made in pr [13665](https://github.com/highfidelity/hifi/pull/13665)
Those changes stop the pelvis from pushing forward when the user leans back. The Fogbugz for that fix is here https://highfidelity.manuscript.com/f/cases/16904/

### Test Plan:
1. open Interface and enter vr mode. press '3' for third person view
2. when you step back your avatar should step backward
3. when you lean back your avatar should match your lean or take a small step back
4. sit in a chair. your avatar should be standing.
5. lean back in the chair and your avatar should lean back slightly or take a step back

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11558/HighFidelity-Beta-PR13692-85b7a9b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11558/HighFidelity-Beta-Interface-PR13692-85b7a9b.exe

**Comment by hyperlogic** [comment]
android and linux build failed for some reason.  test this please

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -3099,9 +3103,12 @@ glm::vec3 MyAvatar::computeCounterBalance() const {
     glm::vec3 counterBalancedCg = (1.0f / DEFAULT_AVATAR_HIPS_MASS) * counterBalancedForHead;
 
     // find the height of the hips
+    const float UPPER_LEG_FRACTION = 0.3333f;
     glm::vec3 xzDiff((cgHeadMass.position.x - counterBalancedCg.x), 0.0f, (cgHeadMass.position.z - counterBalancedCg.z));
     float headMinusHipXz = glm::length(xzDiff);
     float headHipDefault = glm::length(tposeHead - tposeHips);
+    float hipFootDefault = glm::length(tposeHips - tposeRightFoot);
+    float sitSquatThreshold = tposeHips.y - (UPPER_LEG_FRACTION*hipFootDefault);
```
Comment (hyperlogic) [inline interface/src/avatar/MyAvatar.cpp]: spaces between UPPER_LEG_FRACTION * hipFootDefault

File: interface/src/avatar/MyAvatar.cpp [hunk 2]
```diff
@@ -3061,6 +3061,7 @@ glm::vec3 MyAvatar::computeCounterBalance() const {
     JointMass cgRightHandMass(QString("RightHand"), DEFAULT_AVATAR_RIGHTHAND_MASS, glm::vec3(0.0f, 0.0f, 0.0f));
     glm::vec3 tposeHead = DEFAULT_AVATAR_HEAD_POS;
     glm::vec3 tposeHips = glm::vec3(0.0f, 0.0f, 0.0f);
+    glm::vec3 tposeRightFoot = glm::vec3(0.0f, -1.0f, 0.0f);
```
Comment (hyperlogic) [inline interface/src/avatar/MyAvatar.cpp]: use DEFAULT_AVATAR_RIGHTFOOT_POS from AvatarConstants.h

Comment (hyperlogic) [reply interface/src/avatar/MyAvatar.cpp]: Might as well use DEFAULT_AVATAR_HIPS_POS as well.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11591/HighFidelity-Beta-PR13692-85b7a9b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11594/HighFidelity-Beta-PR13692-12cb212.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11594/HighFidelity-Beta-Interface-PR13692-12cb212.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11594/HighFidelity-Beta-PR13692-12cb212.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11594/HighFidelity-Beta-Interface-PR13692-12cb212.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Avatar resets standing position when sitting IRL

**Pull request merged**